// mat-table-exporter/esm2020/lib/mat-table-exporter.directive.mjs
import { Directive, Host, Optional, Self } from '@angular/core';
import { CdkTableExporter } from 'cdk-table-exporter';
import * as i0 from "@angular/core";
import * as i1 from "cdk-table-exporter";
import * as i2 from "@angular/material/table";
export class MatTableExporterDirective extends CdkTableExporter {
    /**
     * Overriding ngAfterViewInit of TableExporter
     */
    ngAfterViewInit() {
        this.exportStarted.subscribe(_ => {
            this.enablePaginator(false);
        });
        this.exportCompleted.subscribe(_ => {
            this.enablePaginator(true);
        });
    }
    constructor(serviceLocator, dataExtractor, table) {
        super(serviceLocator, dataExtractor, table);
    }
    /**
     * MatTable implementation of getPageCount
     */
    getPageCount() {
        const pageSize = this.getPageSize();
        if (pageSize == 0) {
            return 0;
        }
        const pageCount = this.getTotalItemsCount() / this.getPageSize();
        return Math.ceil(pageCount) ?? 1;
    }
    /**
     * MatTable implementation of getPageSize
     */
    getPageSize() {
        return this.getPaginator()?.pageSize ?? 0;
    }
    /**
     * MatTable implementation of getCurrentPageIndex
     */
    getCurrentPageIndex() {
        return this.getPaginator()?.pageIndex ?? 0;
    }
    /**
     * MatTable implementation of getTotalItemsCount
     */
    getTotalItemsCount() {
        return this.getPaginator()?.length ?? this.getDataSource()?.data?.length ?? 0;
    }
    /**
     * MatTable implementation of goToPage
     */
    goToPage(index) {
        const paginator = this.getPaginator();
        paginator.pageIndex = index;
        paginator._changePageSize(paginator.pageSize);
    }
    /**
     * MatTable implementation of getPageChangeObservable
     */
    getPageChangeObservable() {
        return this.getPaginator()?.page;
    }
    getDataSource() {
        return this._cdkTable.dataSource;
    }
    getPaginator() {
        return this.getDataSource()?.paginator;
    }
    enablePaginator(value) {
        const paginator = this.getPaginator();
        paginator.disabled = !value;
        paginator._changePageSize(paginator.pageSize);
    }
}
MatTableExporterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: MatTableExporterDirective, deps: [{ token: i1.ServiceLocatorService }, { token: i1.DataExtractorService }, { token: i2.MatTable, host: true, optional: true, self: true }], target: i0.ɵɵFactoryTarget.Directive });
MatTableExporterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.2.3", type: MatTableExporterDirective, selector: "[matTableExporter]", exportAs: ["matTableExporter"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: MatTableExporterDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[matTableExporter]',
                    exportAs: 'matTableExporter'
                }]
        }], ctorParameters: function () { return [{ type: i1.ServiceLocatorService }, { type: i1.DataExtractorService }, { type: i2.MatTable, decorators: [{
                    type: Host
                }, {
                    type: Self
                }, {
                    type: Optional
                }] }]; } });
