// cdk-table-exporter/esm2020/lib/services/sheetjs-helper.service.mjs
import { Injectable, Inject, Optional } from '@angular/core';
import { XLSX_LIGHTWEIGHT } from '../constants';
import * as i0 from "@angular/core";
export class SheetjsHelperService {
    constructor(xlsxLightweight) {
        this.xlsxLightweight = xlsxLightweight;
    }
    async getXlsx() {
        if (this.xlsxLightweight) {
            return await import(`xlsx/dist/xlsx.mini.min`);
        }
        else {
            return await import('xlsx');
        }
    }
}
SheetjsHelperService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SheetjsHelperService, deps: [{ token: XLSX_LIGHTWEIGHT, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
SheetjsHelperService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SheetjsHelperService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SheetjsHelperService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [XLSX_LIGHTWEIGHT]
                }] }]; } });
