// cdk-table-exporter/esm2020/lib/services/exporters/worksheet-exporter.mjs
import { FileExporter } from './file-exporter';
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../sheetjs-helper.service";
/**
 * An angular service class that is used to create files out of json object arrays.
 */
export class WorksheetExporter extends FileExporter {
    constructor(sheetJsHelper) {
        super();
        this.sheetJsHelper = sheetJsHelper;
    }
    async createContent(rows, options) {
        const workSheet = (await this.sheetJsHelper.getXlsx()).utils.json_to_sheet(rows, {
            skipHeader: true // we don't want to see object properties as our headers
        });
        return await this.workSheetToContent(workSheet, options);
    }
}
WorksheetExporter.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: WorksheetExporter, deps: [{ token: i1.SheetjsHelperService }], target: i0.ɵɵFactoryTarget.Injectable });
WorksheetExporter.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: WorksheetExporter, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: WorksheetExporter, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.SheetjsHelperService }]; } });
