// cdk-table-exporter/esm2020/lib/cdk-table-exporter.module.mjs
import { CdkTableModule } from '@angular/cdk/table';
import { NgModule } from '@angular/core';
import * as i0 from "@angular/core";
export class CdkTableExporterModule {
}
CdkTableExporterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: CdkTableExporterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdkTableExporterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.3", ngImport: i0, type: CdkTableExporterModule, imports: [CdkTableModule] });
CdkTableExporterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: CdkTableExporterModule, imports: [CdkTableModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: CdkTableExporterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [
                        CdkTableModule
                    ],
                    exports: []
                }]
        }] });
